src/storage/mod.rs

```diff
@@ -0,0 +1,56 @@
+//! Storage backend implementations.
```

Comment:
Almost every crate uses a different mod style. Since this is an issue of its own, let split the scope of changes into different PRs and keep the current project structure in this one - focusing only in the storage changes

Reply:
mod.rs is rust 2015 style which we are trying to avoid.

Reply:
Usage the `mod.rs` doesn't mean that you are using an outdated style=) Rust provides an alternative way with `module_name.rs` because it can be easy for Rust newcomers.

I prefer `mod.rs`, because it allows the encapsulation of all the module's stuff in one folder. It is very helpful in cases when you have many modules. Because it removes "dust" files from the top of the crate.

As an example, `fuel-core/fuel-core/src`:

<img width="368" alt="image" src="https://user-images.githubusercontent.com/18346821/188751231-0ed973ab-63b8-40f7-99b1-fd5109a5aa88.png">

It can be simplified into:
<img width="345" alt="image" src="https://user-images.githubusercontent.com/18346821/188753049-3c8f9890-16b9-4a07-a030-91ef78616413.png">

You can see that top of the `src` folder with the `mod.rs` style contains files only with custom logic not related to any modules(they belong only to `src`). It helps you better understand the relationship of files/modules with one look at the folder tree.

Of course, it is only my preference, and if everybody likes the approach with `module_name.rs`, I will use it=) But I don't see pros using `module_name.rs` for crates with many modules.

Reply:
This is bikeshedding and we've already adopted the rust 2018 style, I don't really want to debate all these preferences constantly.